core/src/main/java/io/fabric8/maven/core/service/kubernetes/BuildahIntegration/BuildahBuildServiceUtil.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.service.kubernetes.BuildahIntegration;
+
+import io.fabric8.maven.core.service.BuildService;
+import io.fabric8.maven.core.util.FatJarDetector;
+import io.fabric8.maven.docker.access.AuthConfig;
+import io.fabric8.maven.docker.config.BuildImageConfiguration;
+import io.fabric8.maven.docker.service.RegistryService;
+import io.fabric8.maven.docker.util.Logger;
+import io.jshift.buildah.core.Buildah;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class BuildahBuildServiceUtil {
+
+
+    public static void buildImage(BuildahBuildConfiguration buildConfiguration, Logger log) {
+
+        String fromImage = buildConfiguration.getFrom();
+        String targetImage = buildConfiguration.getTargetImage();
+        Map<String, String> envMap  = buildConfiguration.getEnvMap();
+        Map<String, String> labelMap  = buildConfiguration.getLabelMap();
+        List<String> envList = null;
+        List<String> labelList = null;
+        if(envMap != null) {
+            envList = envList(envMap);
+        }
+
+        if(labelMap != null) {
+            labelList = labelList(labelMap);
+        }
+
+        List<String> entrypointList = new ArrayList<>();
+        if(buildConfiguration.getEntryPoint() != null) {
+            entrypointList = buildConfiguration.getEntryPoint().asStrings();
+        }
+
+        List<String> portList = buildConfiguration.getPorts();
+        String targetDir = buildConfiguration.getTargetDir();
+        Path fatJar = buildConfiguration.getFatJar();
+        AuthConfig authConfig = buildConfiguration.getAuthConfig();
+
+        buildImage(fromImage, targetImage, envList, labelList, portList, fatJar, targetDir, log, entrypointList);
+    }
+
+    public static List<String> envList(Map<String, String> envMap) {
+        List<String> list = new ArrayList<>();
+
+        for(String key : envMap.keySet()) {
+            String value = envMap.get(key);
+            String envVar = key + "=" + value;
+            list.add(envVar);
+        }
+        return list;
+    }
+
+    public static List<String> labelList(Map<String, String> labelMap) {
+        List<String> list = new ArrayList<>();
```

Comment:
both of these methods seem to be doing the same thing. Can we combine these into one?